kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/appliers/JvmTestTaskConfigurator.kt

```diff
@@ -39,6 +39,12 @@ internal class JvmTestTaskConfigurator(
             }
         testTask.dependsOn(data.findAgentJarTask)
 
+        testTask.doFirst {
+            // delete report so that when the data is re-measured, it is not appended to an already existing file
+            // see https://github.com/Kotlin/kotlinx-kover/issues/489
+            binReportProvider.get().asFile.delete()
```

Comment:
Does it matter if it was in the build cache or not?

Reply:
No difference, because `doFirst` is executed only when the Gradle task is running (it has the SUCCESS status).
If the task has the status of `UP-TO-DATE` or `FROM-CACHE`, then `doFirst` is not called